storage/HashTable.hpp

```diff
@@ -1429,6 +1492,11 @@ HashTablePutResult HashTable<ValueT, resizable, serializable, force_key_copy, al
                                        variable_size,
                                        (*functor)(*accessor),
                                        using_prealloc ? &prealloc_state : nullptr);
+            // Insert into bloom filter, if enabled.
+            if (has_build_side_bloom_filter_) {
+              this->build_bloom_filter_->insert(static_cast<const std::uint8_t *>(key.getDataPtr()),
```

Comment:
I see, thanks for that information. However, we are inserting an entry in the bloom filter for every tuple, and while doing so, acquiring a lock on the bloom filter. Even though the time to insert an entry in the bloom filter may not be high (flipping a single bit), we are essentially serializing all the worker threads building the hash table. 

I feel that there are some relatively easy optimizations possible, and it may be worth discussing them.